bugfix/empty_tip_root

v8p1197 opened the pull request:
Closes #9.

v8p1197 committed:
[FIX] Handle empty root and tip parameters

- [ADD] With empty root, defaults to chain root
- [ADD] With empty tip, raises an exception
- [TST] Add two test fixtures to test the two scenarios above

v8p1197 committed:
Merge branch 'main' into bugfix/empty_tip_root

v8p1197 committed:
[REF] Reorder dependencies alphabetically

eferre89 commented on kdl_inverse_dynamics_solver/src/kdl_inverse_dynamics_solver.cpp:
```diff
@@ -66,25 +66,29 @@ void InverseDynamicsSolverKDL::initialize(rclcpp::node_interfaces::NodeParameter
   {
     parameters_interface->get_parameter(ns + "root", root_param);
     root = root_param.as_string();
+    root_found = !root.empty();
```
In my opinion, the case "the user provides an empty `root` parameter" and the case "the user does not provide a `root` parameter" are very different, and should be handled differently. Indeed, in the former case, the user knows that a `root` parameter is required, but they provide the empty string on purpose. In this case, the class shall not assume that the user wants to use the chain's root segment. For example, in developing an application, the user might want to decide the root later on, "stubbing" it to `""` temporarily. In the latter case, either the user does not know that a root parameter is required (in this case, assuming the root segment sounds reasonable) or they know that the application will use the root segment by default.

v8p1197 replied:
Ok, the scenarios you are describing are intriguing and sound reasonable.

To correctly implement them, the class should expose an interface to reconfigure the filter (or, at least, the root link parameter), to support the "set the root link later on" use case.
Do you agree with this choice?
Shall I implement this?

v8p1197 replied:
In an internal meeting, we agreed to just log different messages, to warn the user about the usage of this module via either warnings or exceptions.

eferre89 commented on kdl_inverse_dynamics_solver/src/kdl_inverse_dynamics_solver.cpp:
```diff
@@ -66,25 +66,29 @@ void InverseDynamicsSolverKDL::initialize(rclcpp::node_interfaces::NodeParameter
-  else
+  if (!tip_found)
   {
     throw inverse_dynamics_solver::ParameterUninitializedException("Failed to find parameter 'tip'.");
```
This exception message does not match the actual error condition: we are telling the user that the parameter `tip` does not exist, while it might be empty. The two conditions shall be handled separately.

v8p1197 replied:
That's fair. I'll do that.

v8p1197 committed:
[REF] Manages missing/empty root/tip parameters with different logics

- [TST] Add related tests

eferre89 committed:
[REF] Use param_namespace instead of ns in warning/error messages to print the namespace without a trailing dot

- [DOC] Fix Doxygen and inline documentation to comply to the last implementation and fix typos
